apollo-federation/src/query_plan/fetch_dependency_graph.rs

```diff
@@ -1158,6 +1286,339 @@ impl FetchDependencyGraph {
         // And so it is a sequence, even if the roots will be queried in parallel.
         Ok((processor.reduce_sequence(main_sequence), deferred))
     }
+
+    fn can_merge_child_in(
+        &self,
+        node_id: NodeIndex,
+        child_id: NodeIndex,
+    ) -> Result<bool, FederationError> {
+        let node = self.node_weight(node_id)?;
+        let child = self.node_weight(child_id)?;
+        let parent_relation = self.parent_relation(child_id, node_id);
+
+        Ok(node.subgraph_name == child.subgraph_name
```

Comment:
Yes... and also due to the fact that in RS we operate on IDs and need to fetch the node to get actual subgraph names so instead of doing the lookups twice I just moved it here